developer/src/kmcmplib/src/Compiler.cpp

```diff
@@ -2413,166 +2399,172 @@ KMX_DWORD GetXStringImpl(PKMX_WCHAR tstr, PFILE_KEYBOARD fk, PKMX_WCHAR str, KMX
+          if (*q != chQuote)
+            return KmnCompilerMessages::ERROR_InvalidToken;
+          q++; // skip quote
+        } else {
+          for (j = 0; !iswspace(*q) && *q != ']' && *q != 0; q++, j++);
 
-        KMX_WCHAR vkname[SZMAX_VKDICTIONARYNAME];  // I3438
+          if (*q == 0)
+            return KmnCompilerMessages::ERROR_InvalidToken;
 
-        if (j >= SZMAX_VKDICTIONARYNAME)
-          return KmnCompilerMessages::ERROR_InvalidToken;
+          KMX_WCHAR vkname[SZMAX_VKDICTIONARYNAME];  // I3438
 
-        u16ncpy(vkname, p, j);  // I3481
-        vkname[j] = 0;
+          if (j >= SZMAX_VKDICTIONARYNAME)
+            return KmnCompilerMessages::ERROR_InvalidToken;
 
-        if (u16icmp(vkname, u"K_NPENTER") == 0)
-          i = 5;  // I649 - K_NPENTER hack
-        else {
-          for (i = 0; i <= VK__MAX; i++) {
-            if (u16icmp(vkname, VKeyNames[i]) == 0 || u16icmp(vkname, VKeyISO9995Names[i]) == 0)
-              break;
-          }
-        }
+          u16ncpy(vkname, p, j);  // I3481
+          vkname[j] = 0;
 
-        if (i == VK__MAX + 1) {
-          if(!VerifyKeyboardVersion(fk, VERSION_90)) {
-            return KmnCompilerMessages::ERROR_InvalidToken;
+          if (u16icmp(vkname, u"K_NPENTER") == 0)
+            i = 5;  // I649 - K_NPENTER hack
+          else {
+            for (i = 0; i <= VK__MAX; i++) {
+              if (u16icmp(vkname, VKeyNames[i]) == 0 || u16icmp(vkname, VKeyISO9995Names[i]) == 0)
+                break;
+            }
           }
 
-          i = GetVKCode(fk, vkname);  // I3438
-          if (i == 0)
-            return KmnCompilerMessages::ERROR_InvalidToken;
-        }
+          if (i == VK__MAX + 1) {
+            if(!VerifyKeyboardVersion(fk, VERSION_90)) {
+              return KmnCompilerMessages::ERROR_InvalidToken;
+            }
 
-        tstr[mx++] = (int)i;
+            i = GetVKCode(fk, vkname);  // I3438
+            if (i == 0)
+              return KmnCompilerMessages::ERROR_InvalidToken;
+          }
+
+          tstr[mx++] = (int)i;
 
-        if (kmcmp::FMnemonicLayout && (i <= VK__MAX) && VKeyMayBeVCKey[i]) {
-          ReportCompilerMessage(KmnCompilerMessages::WARN_VirtualKeyWithMnemonicLayout);  // I3438
+          if (kmcmp::FMnemonicLayout && (i <= VK__MAX) && VKeyMayBeVCKey[i]) {
+            ReportCompilerMessage(KmnCompilerMessages::WARN_VirtualKeyWithMnemonicLayout);  // I3438
+          }
         }
-      }
 
-      while (iswspace(*q)) q++;
+        while (iswspace(*q)) q++;
 
-      if (*q != ']')
-        return KmnCompilerMessages::ERROR_InvalidToken;
+        if (*q != ']')
+          return KmnCompilerMessages::ERROR_InvalidToken;
 
-      tstr[mx++] = UC_SENTINEL_EXTENDEDEND;
-      tstr[mx] = 0;
-      //printf("--EXTENDEDEND--\n");
+        tstr[mx++] = UC_SENTINEL_EXTENDEDEND;
+        tstr[mx] = 0;
+        //printf("--EXTENDEDEND--\n");
 
-      p = q + 1;
+        p = q + 1;
 
-      sFlag = 0;
+        sFlag = 0;
```

Comment:
```suggestion
```

nit only :grin:

Reply:
The more unneeded code removed the better.

Reply:
LGTM also